### PR TITLE
Feature/kinetic support

### PR DESCRIPTION
**Pull request by agutenkunst**
## Description

Allows building under ROS kinetic.

Remarks:
-  Switched (back) to rosfmt which works fine for kinetic/melodic/noetic
- Excluded one integration-test, since it needs a feature of googletest not present in kinetic
- To stop `catkin_lint` from failing due to the usage of `ENV{ROSDISTRO}` in the CMakelists `--ignore env_var` is needed.
- Extra braces `{` are needed for gcc 5.4

- [x] cherry-pick #272
- [ ] Hold a few days until decision by @jschleicher if this should be a sperate branch or merged into main
---

<details>
  <summary>PR Checklist</summary>

### Things to add, update or check by the maintainers before this PR can be merged.

- [ ] Public api function documentation
- [ ] Architecture documentation reflects actual code structure
- [ ] [Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)
- [ ] Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)
- [ ] Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))
- [ ] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
- [ ] CHANGELOG.rst updated
- [ ] Copyright headers
- [ ] Examples

### Review Checklist
- [ ] Soft- and hardware architecture (diagrams and description)
- [ ] Test review (test plan and individual test cases)
- [ ] Documentation describes purpose of file(s) and responsibilities
- [ ] Code (coding rules, style guide)

### Release planning (please answer)
- [ ] When is the new feature released?
- [ ] Which dependent packages do have to be released simultaneously?

### Hardware tests
_Unstrike the text below to enable automatic hardware tests if available. Otherwise use this as a request for the reviewer._
- [ ] ~Perform hardware tests~

</details>

### CODE REVIEW

**Comment by rfeistenauer** [comment]
I think its notifyable that building the package with catkin and rosfmt installed but not used can cause fmt errors. 
So If we include this into main and therefore all install rosfmt we will have to remember to deinstall rosfmt if we want to build older versions of the driver.
It seems however not to intervene with the standalone build.
Maybe we should look into this if we plan to include this to main.

```
/usr/bin/ld: CMakeFiles/psen_scan_v2_node.dir/src/psen_scan_driver.cpp.o: in function `main':
psen_scan_driver.cpp:(.text.startup+0x1021): undefined reference to `fmt::v7::detail::vformat[abi:cxx11](fmt::v7::basic_string_view<char>, fmt::v7::format_args)'
/usr/bin/ld: psen_scan_driver.cpp:(.text.startup+0x106b): undefined reference to `fmt::v7::detail::vformat[abi:cxx11](fmt::v7::basic_string_view<char>, fmt::v7::format_args)'
/usr/bin/ld: CMakeFiles/psen_scan_v2_node.dir/src/psen_scan_driver.cpp.o: in function 
...
```

**Review by rfeistenauer — APPROVED** [review]
Generally I approve, but we should take a closer look at why we have those conflicts between fmt and rosfmt if we really consider taking this into main rather than on a separate branch

File: .github/workflows/ci_action_kinetic.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  ADDITIONAL_DEBS: "clang-tidy libclang-dev"
+  CATKIN_LINT: true
+  CATKIN_LINT_ARGS: '--strict --ignore env_var'
+  CMAKE_ARGS: "-DCATKIN_ENABLE_CLANG_TIDY=true"
+  ROS_DISTRO: kinetic
+
+jobs:
+  testing-repo:
+    name: "Build + Test with Testing Repo of kinetic (http://packages.ros.org/ros-testing/ubuntu)"
+    runs-on: ubuntu-18.04
```
Comment (martiniil) [inline .github/workflows/ci_action_kinetic.yml:15]: Is this intential? kinetic runs on 16.04 @agutenkunst

Comment (jschleicher) [reply .github/workflows/ci_action_kinetic.yml:15]: As far as I remember, the appropriate docker image is selected based on `ROS_DISTRO` - regardless of the host OS.

**Comment by agutenkunst** [comment]
Closed use https://github.com/PilzDE/psen_scan_v2/tree/feature/kinetic-support

**Pull request closed without merging**